src/vmm/src/guest_config/cpuid/amd/mod.rs

```diff
@@ -56,6 +58,34 @@ impl From<AmdCpuid> for RawCpuid {
     }
 }
 
+impl From<kvm_bindings::CpuId> for AmdCpuid {
+    #[inline]
+    fn from(kvm_cpuid: kvm_bindings::CpuId) -> Self {
+        let map = kvm_cpuid
+            .as_slice()
+            .iter()
+            .map(|entry| {
+                (
+                    CpuidKey {
+                        leaf: entry.function,
+                        subleaf: entry.index,
+                    },
+                    CpuidEntry {
+                        flags: KvmCpuidFlags(entry.flags),
+                        result: CpuidRegisters {
+                            eax: entry.eax,
+                            ebx: entry.ebx,
+                            ecx: entry.ecx,
+                            edx: entry.edx,
+                        },
+                    },
+                )
+            })
+            .collect();
```

Comment:
Unfortunately it seems to be difficult to do it straightforward.
```
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
   --> src/vmm/src/guest_config/cpuid/mod.rs:424:1
    |
424 | impl From<kvm_bindings::kvm_cpuid_entry2> for (CpuidKey, CpuidEntry) {
    | ^^^^^------------------------------------^^^^^----------------------
    | |    |                                        |
    | |    |                                        this is not defined in the current crate because tuples are always foreign
    | |    `kvm_cpuid_entry2` is not defined in the current crate
    | impl doesn't use only types from inside the current crate
    |
    = note: define and implement a trait or new type instead

For more information about this error, try `rustc --explain E0117`.
warning: `vmm` (lib) generated 2 warnings
error: could not compile `vmm` due to previous error; 2 warnings emitted
```

Reply:
Do you think it is better to define a new type wrapping `(CpuidKey, CpuidEntry)` to do this?

Reply:
Hmm, I forgot about this, no I wouldn't think its worth defining a wrapping type to do this.

Reply:
Got it! Actually I didn't think about it, thanks for your good input!

Reply:
If you don't have any concerns, I'd appreciate it if you could hit "approve" button!